### ISSUE TITLE
"Incompatible", "Requires armeabi-v7a"

### ISSUE THREAD

**Issue by RedSteel-1** [darkmoonight/Rain#112]
"Incompatible", "Requires armeabi-v7a" - this is what I see in my software store with Izzy repo included.
So stuck with v1.2.2 and not able to upgrade to 1.2.5.

Can this be fixed?

**Comment by Leonavichus** [comment]
@IzzySoft, can you help me figure this out?

**Comment by IzzySoft** [comment]
We've set up armeabi-v7a here, that's correct. So if the device (which @A-7666 didn't mention) does not support that ABI, the app cannot be updated via my repo (and probably wasn't installed from there in the first place – can you confirm?)

I assume the device is not supporting the ABI – in which case we cannot fix that here without breaking it for other users whose devices need this very ABI. I can only support one ABI in my repo, sorry.

**Comment by IzzySoft** [comment]
Right, the Pixel 7 pro dropped 32bit support. So you cannot have installed the app from my repo, which always only had the 32bit version.

**Comment by IzzySoft** [comment]
> Could you include both 32 and 64bit in your repo?

Nope, my framework does not support that. It would then mean I had to manually update all those apps, which I lack the time for.

> But, this is not true.

Sorry to contradict you, but it is. I cannot remember a single case where I switched an app from arm64 to arm32.

> There are plenty of programs I have installed from Izzy-repo and they all have either only64bit or both 32/64bit available.

That is true, but not for this app. And some apps support multiple architectures in a single APK (before you ask: no, I would not recommend that for Rain as it would then soon reach the size limits and I could no longer offer it at all).

Sorry, but it does not help to push further, @A-7666 – that would not change a thing: I cannot support multiple APKs per version in my repo. Your device is the only one having this issue, all other 64bit devices I know of support 32bit APKs as well – but the other way around is not true for any device (no 32bit device can support 64bit apps, so switching to 64bit in my repo would make the app unavailable for **all** 32bit devices as opposed to just the Pixel 7 pro).

The alternative would be bringing Rain to F-Droid.org (guess that's what #22 is about), where per-ABI releases are supported.
